libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/component/maneuver/ManeuverViewBinder.kt

```diff
@@ -20,6 +26,9 @@ class ManeuverViewBinder : UIBinder {
         TransitionManager.go(scene)
         val binding = MapboxManeuverGuidanceLayoutBinding.bind(value)
 
-        return ManeuverComponent(binding.maneuverView)
+        return reloadOnChange(loadedMapStyle) {
+            if (it != null) ManeuverComponent(binding.maneuverView, it)
+            else EmptyComponent()
```

Comment:
small comment: We could have `reloadOnChange` take `null`. feels simpler to me considering the return type is `MapboxNavigationObserver`

```suggestion
            else null
```

``` kotlin
fun <T> reloadOnChange(value: Flow<T>, factory: (value: T) -> UIComponent): UIComponent =
    object : UIComponent() {
        private var childComponent: UIComponent? = null
```

Reply:
Otherwise instead of introducing the empty component, could it use NoOpMapboxNavigationObserver?

Reply:
Done